### PR TITLE
add serviceMonitor for proxy metrics

### PR DESCRIPTION
**Pull request by ranakan19**
changes wrt https://issues.redhat.com/browse/ASC-248

### CODE REVIEW

**Review by xcoulon — APPROVED** [review]

**Review by bamachrn — APPROVED** [review]

File: components/monitoring/prometheus/base/servicemonitors/sandbox-registration-service-proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    interval: 15s
+    scheme: http
+    path: /metrics
+    port: proxy-metrics
+  namespaceSelector:
+    matchNames:
+      - toolchain-host-operator
+  selector:
+    matchLabels:
+      run: proxy-metrics
```
[the remark this answers is not in the document]
Comment (ranakan19) [reply components/monitoring/prometheus/base/servicemonitors/sandbox-registration-service-proxy.yaml]: Thanks, added in [189c96a](https://github.com/redhat-appstudio/infra-deployments/pull/2577/commits/189c96ae2292b44c5e02b1958529f0daefeb18ac)

File: components/sandbox/base/monitoring/sandbox-registration-service-proxy.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  namespace: toolchain-host-operator
+  name: sandbox-registration-service-proxy
+  labels:
+    prometheus: appstudio-workload
+spec:
+  endpoints:
+    interval: 15s
+    scheme: http
+    path: /metrics
+    port: proxy-metrics
```
Comment (yftacherzog) [inline components/sandbox/base/monitoring/sandbox-registration-service-proxy.yaml:13]: This should be something like this I suspect:
```suggestion
  endpoints:
    - interval: 15s
      scheme: http
      path: /metrics
      port: proxy-metrics
```

File: components/sandbox/base/monitoring/sandbox-registration-service-proxy.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  namespaceSelector:
+    matchNames:
+      - toolchain-host-operator
```
Comment (yftacherzog) [inline components/sandbox/base/monitoring/sandbox-registration-service-proxy.yaml:16]: I don't think you need the namespace selector.

**Review by yftacherzog — CHANGES_REQUESTED** [review]
I still think there need to be a few changes for this to work

File: components/sandbox/base/monitoring/sandbox-registration-service-proxy.yaml (hunk 3 above)
Comment (ranakan19) [reply components/sandbox/base/monitoring/sandbox-registration-service-proxy.yaml]: done in [d293689](https://github.com/redhat-appstudio/infra-deployments/pull/2577/commits/d293689b2791c4af598a7bb70759b9505c037b11)

**Review by xcoulon — APPROVED** [review]

**Comment by yftacherzog** [comment]
Could you rebase and squash it to a single commit?

**Comment by ranakan19** [comment]
> quash it to a single commit?

this PR keeps getting out of date with the base branch very quickly and its hard to keep up since every time i update branch its a new commit. I usually do the squash and merge option to merge. Is merging automated?

**Comment by ranakan19** [comment]
/retest 
some cluster operators are not available

**Comment by ranakan19** [comment]
@yftacherzog Could you take another look? I've addressed all your comments

**Review by yftacherzog — APPROVED** [review]

**Pull request merged**